AutoSaveFile/AutoSaveFilePackage.cs

```diff
@@ -157,18 +174,27 @@ private void CancelPreviousSaveTask()
             }
         }
 
-        private static void SaveDocument(DTE dte)
+        private bool ShouldSaveDocument(Window window)
         {
-            var window = dte.ActiveWindow;
             var windowType = window.Kind;
 
             if (windowType == "Document")
             {
-                window.Project?.Save();
-                window.Document?.Save();
+                var fileType = _helper.GetFileType(window);
+                var optionsPage = (OptionPageGrid)GetDialogPage(typeof(OptionPageGrid));
+                var ignoredFileTypes = optionsPage.IgnoredFileTypes?.Split(',')
+                    .Select(str => str.Trim());
+
+                if (ignoredFileTypes != null && !ignoredFileTypes.Contains(fileType))
+                {
+                    return true;
+                }

```

Comment:
About 'optionsPage.IgnoredFileTypes', if the user didn't input anything or the text block in the option page is empty, does this returns a null or an empty string?
If the local variable 'ignoredFileTypes' evaluated to null than the whole function will return false resulting in an unwanted behavior. (the user did not necessarily want to ignore anything but the application will always ignore because 'ignoredFileTypes' is null)